spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/hazelcast/HazelcastAutoConfigurationServerTests.java

```diff
@@ -59,38 +59,63 @@ public void defaultConfigFile() {
 	}
 
 	@Test
-	public void systemProperty() {
-		this.contextRunner
-				.withSystemProperties(HazelcastServerConfiguration.CONFIG_SYSTEM_PROPERTY
-						+ "=classpath:org/springframework/boot/autoconfigure/hazelcast/hazelcast-specific.xml")
-				.run((context) -> {
-					Config config = context.getBean(HazelcastInstance.class).getConfig();
-					assertThat(config.getQueueConfigs().keySet()).containsOnly("foobar");
-				});
+	public void systemPropertyWithXml() {
+		systemProperty(HazelcastServerConfiguration.CONFIG_SYSTEM_PROPERTY
+				+ "=classpath:org/springframework/boot/autoconfigure/hazelcast/hazelcast-specific.xml");
+	}
+
+	@Test
+	public void systemPropertyWithYaml() {
+		systemProperty(HazelcastServerConfiguration.CONFIG_SYSTEM_PROPERTY
+				+ "=classpath:org/springframework/boot/autoconfigure/hazelcast/hazelcast-specific.yaml");
+	}
+
+	private void systemProperty(String systemProperties) {
+		this.contextRunner.withSystemProperties(systemProperties).run((context) -> {
+			Config config = context.getBean(HazelcastInstance.class).getConfig();
+			assertThat(config.getMapConfigs().keySet()).containsOnly("foobar");
+		});
 	}
 
 	@Test
-	public void explicitConfigFile() {
-		this.contextRunner.withPropertyValues(
+	public void explicitConfigFileWithXml() {
+		explicitConfigFile(
 				"spring.hazelcast.config=org/springframework/boot/autoconfigure/hazelcast/"
-						+ "hazelcast-specific.xml")
-				.run((context) -> {
-					Config config = context.getBean(HazelcastInstance.class).getConfig();
-					assertThat(config.getConfigurationFile())
-							.isEqualTo(new ClassPathResource(
-									"org/springframework/boot/autoconfigure/hazelcast"
-											+ "/hazelcast-specific.xml").getFile());
-				});
+						+ "hazelcast-specific.xml");
 	}
 
 	@Test
-	public void explicitConfigUrl() {
+	public void explicitConfigFileWithYaml() {
+		explicitConfigFile(
+				"spring.hazelcast.config=org/springframework/boot/autoconfigure/hazelcast/"
+						+ "hazelcast-specific.yaml");
+	}
+
+	private void explicitConfigFile(String propertyValues) {
+		this.contextRunner.withPropertyValues(propertyValues).run((context) -> {
+			Config config = context.getBean(HazelcastInstance.class).getConfig();
+			assertThat(config.getMapConfigs().keySet()).containsOnly("foobar");
+		});
+	}
+
+	@Test
+	public void explicitConfigUrlWithXml() {
+		explicitConfigUrl("hazelcast-default.xml");
+	}
+
+	@Test
+	public void explicitConfigUrlWithYaml() {
+		explicitConfigUrl("hazelcast-default.yaml");
+	}
+
+	private void explicitConfigUrl(String configFileName) {
 		this.contextRunner
-				.withPropertyValues("spring.hazelcast.config=hazelcast-default.xml")
+				.withPropertyValues(
+						String.format("spring.hazelcast.config=%s", configFileName))
```

Comment:
Why String.format here but not in the others?

Reply:
changed